src/workerd/api/http.h

```diff
@@ -639,14 +662,51 @@ class Request: public Body {
   jsg::Optional<v8::Local<v8::Object>> getCf(jsg::Lock& js);
   // Returns the `cf` field containing Cloudflare feature flags.
 
-  jsg::WontImplement getContext() { return jsg::WontImplement(); }
-  // This is deprecated in the spec.
-
-  jsg::WontImplement getMode()        { return jsg::WontImplement(); }
-  jsg::WontImplement getCredentials() { return jsg::WontImplement(); }
-  jsg::Unimplemented getIntegrity()   { return jsg::Unimplemented(); }
-  jsg::Unimplemented getCache()       { return jsg::Unimplemented(); }
-  // See members of Initializer for commentary on unimplemented APIs.
+  bool getKeepalive() { return false; }
+
+  v8::Local<v8::Value> getDuplex(jsg::Lock& js) { return js.v8Undefined(); }
+  // The duplex option controls whether or not a fetch is expected to send the entire request
+  // before processing the response. The default value ("half"), which is currently the only
+  // option supported by the standard, dictates that the request is fully sent before handling
+  // the response. There are currently a proposal to add a "full" option which is the model
+  // we support. Once "full" is added, we need to update this to accept either undefined or
+  // "full", and possibly decide if we want to support the "half" option. For now, we'll
+  // explicitly return undefined for this property.
+
+  v8::Local<v8::Value> getContext(jsg::Lock& js) { return js.v8Undefined(); }
+  // This is deprecated in the spec and has since been removed from the spec
+  // entirely. In discussions with other implementers, it was determined that
+  // simply returning undefined for these kinds of unsupported old properties
+  // was preferred to throwing.
+  // TODO(cleanup) Alternatively, we might just decide to not have this property
+  // at all.
+
+  v8::Local<v8::Value> getMode(jsg::Lock& js) { return js.v8Undefined(); }
```

Comment:
Instead of returning `undefined`, should we just not declare these properties at all? Referencing them will still produce `undefined` either way, but declaring the properties on the prototype sort of makes it look like they might be implemented.

Reply:
I'd rather declare them so that they will pass checks like `if ('mode' in response) {...}`. In a sense we are implementing these in the sense that we are *explicitly* saying yes, we know this standard property exists but we do not support it at all. But I don't feel super strongly about it.

Reply:
I'll go ahead and remove them.

Reply:
oh, the other thing that specifying these does is prevent users from hacking it, leading to a potential source of breakage later if/when we do choose to implement it... for instance, with these defined as readonly prototype properties, if someone in their code did `request.duplex = 'whatever'`, setting the value would be ignored and the value would always return as `undefined`. In a sense, it reserves the usage of the property for later implementation.